src/pod.ts

```diff
@@ -1753,4 +1747,21 @@ export class PodConnections {
   private extractPorts(selector?: networkpolicy.INetworkPolicyPeer): networkpolicy.NetworkPolicyPort[] {
     return container.extractContainerPorts(selector).map(n => networkpolicy.NetworkPolicyPort.tcp(n.number));
   }
+
+  /**
+   * Sets the default network policy for Pods to have all egress and ingress connections as disabled
+   */
+  private defaultNetworkPolicy() {
+    new networkpolicy.NetworkPolicy(this.instance, 'DefaultDenyAll', {
+      // the policy must be defined in the namespace of the pod
+      // so it can select it.
+      metadata: Lazy.any({ produce: () => {return { namespace: this.instance.metadata.namespace };} }),
+      egress: {
+        default: networkpolicy.NetworkPolicyTrafficDefault.DENY,
+      },
+      ingress: {
+        default: networkpolicy.NetworkPolicyTrafficDefault.DENY,
+      },
+    });
```

Comment:
So for every pod in the same namespace, this will create the same network policy..a bit awkward.
Also, this wasn't the intent. I can see how this got mixed up because the issue didn't describe it well enough. 

The idea was to allow the user to easily create an isolated **pod**, not an entire **namespace**.
I do think namespace isolation is a cool feature as well, but this should be implemented in the [Namespace](https://github.com/cdk8s-team/cdk8s-plus/blob/k8s-24/main/src/namespace.ts) construct, not the pod.

Reply:
I understand, will make the update.